lib/vyasa/written.ex

```diff
@@ -37,6 +52,54 @@ defmodule Vyasa.Written do
   """
   def get_text!(id), do: Repo.get!(Text, id)
 
+  @doc """
+  Gets a single source by id.
+
+  Raises `Ecto.NoResultsError` if the Text does not exist.
+
+  ## Examples
+
+      iex> get_source!(<uuid>)
+      %Source{}
+
+      iex> get_text!(<uuid>)
+      ** (Ecto.NoResultsError)
+
+  """
+  def get_source!(id), do: Repo.get!(Source, id)
+  |> Repo.preload(:chapters)
+  |> Repo.preload(:verses)
+
+  def get_source_by_title(title) do
+    list_sources()
+    |>Enum.find(fn src -> src.title == title end)
```

Comment:
I face an issue trying to change the function to this, probably just a syntax thing


```elixir
  def get_source_by_title(title) do
    query = from src in "sources", # point 1
    # query = from src in {"public", "sources"}, #point 2
    # query = from src in Source, # point 3
            where: src.title ==  ^title,
            select: src

    Repo.one(query)
    |> preload([:chapters, :verses])
  end


```


<details><summary>1. ref docs, the table name just supplied as a string, as per schema name defined in the schema module ==> but this will give me the error of "no schema defined" </summary>

ref the [table name just supplied as a string, as per schema name defined in the schema module](https://hexdocs.pm/ecto/Ecto.Query.html#summary)

![image](https://github.com/ve1ld/vyasa/assets/38996397/17cf18dd-29da-44db-b2d1-a1fd9116eb11)


</details> 

<details><summary>2. I follow the error, try adding in schema name via a tuple, it continues giving me issues</summary>

Likely because the guard for that fuction is: 

`def to_query({source, schema} = from) when is_binary(source) and is_atom(schema) and not (schema == nil)`

![image](https://github.com/ve1ld/vyasa/assets/38996397/5c3c702f-a0ff-4bcc-90bb-355ce947edef)

</details> 

<details><summary>3. So I try to follow the usage in Orbius repo, where it directly alises the module and uses that instead of a string... not successful yet</summary>

![image](https://github.com/ve1ld/vyasa/assets/38996397/53dc29f6-d3d4-461b-b630-5c804b9c883e)


</details> 



it's definitely some minor syntax issue, gonna move on first, will ask over call

Reply:
Found the way to do this, instead of passing a string, have to properly define the particular schema module 

e.g. 

```elixir

import Ecto.Query

Vyasa.Repo.all(Vyasa.Written.Verse)
|> Repo.preload([:translations, :source, :chapter])
|> Enum.filter(fn v -> is_nil(v.global_order) end)

query = from v in Vyasa.Written.Verse,
        where: v.no == 1,
        where: v.chapter_no == 5,
        select: v
res = Repo.all(query)
```

Reply:
I'll come back to touching this up in a bit

Reply:
done where I could 

in other areas, where it involves querying then piping it such that we query on an association from the previous query, i left that part out cuz i'm behind time now 

the suggestion on query builder pattern has been skipped as well, teach me quickly if it's something trivial for you